core/primitives/src/sharding.rs

```diff
@@ -117,7 +117,8 @@ pub struct ChunkOnePart {
 }
 
 #[derive(BorshSerialize, BorshDeserialize, Debug, Clone, Eq, PartialEq)]
-pub struct ShardProof(pub ShardId, pub MerklePath);
+// from_shard_id -> to_shard_id
+pub struct ShardProof(pub ShardId, pub ShardId, pub MerklePath);
```

Comment:
make the fields named? it's already a struct - naming make sense.

Reply:
++